File: problem001/euler1.java
import java.text.DecimalFormat;

public class euler1 {
	public static void main(String[] args) {
		double sum;
		int n = 999;
		DecimalFormat formatter = new DecimalFormat("#0");
		sum = sum_n(n, 3) + sum_n(n, 5) - sum_n(n, 15);
		System.out.println(formatter.format(sum));
	}
	private static double sum_n(int n, int mul) {
		double sum;
		sum = (n - (n % mul) + mul) * (n / mul / 2.0);
		return sum;
	}
}

File: problem001/euler1.scala
object euler1 {

    def sum_m(n: Int, mul: Int) : Double = {
        return (n - (n % mul) + mul) * (n / mul / 2.0)
    }

    def main(args: Array[String]) {
        val n = 999
        println(sum_m(n, 3) + sum_m(n, 5) - sum_m(n, 15))
    }
}
